skyportal/plot.py

```diff
@@ -687,7 +687,212 @@ def photometry_plot(obj_id, user, width=600, height=300):
 
     p2 = Panel(child=layout, title='Mag')
 
-    tabs = Tabs(tabs=[p2, p1], width=width, height=height, sizing_mode='fixed')
+    # now make period plot
+
+    if obj.varstar:
+
+        # get period from annotations (any for now)
+        annotation_list = obj.get_annotations_readable_by(user)[0]
+        if 'period' in annotation_list.data:
+            period = annotation_list.data['period']
+        else:
+            period = 0.1
+
+        # bokeh figure for period plotting
+        period_plot = figure(
+            aspect_ratio=1.7,
+            sizing_mode='scale_both',
+            active_drag='box_zoom',
+            tools='box_zoom,wheel_zoom,pan,reset,save',
+            y_range=(ymax, ymin),
+            x_range=(-0.1, 2.1),  # cover two phases: a and b
+            toolbar_location='above',
+            toolbar_sticky=False,
+            x_axis_location='below',
+        )
+
+        # axis labels
+        period_plot.xaxis.axis_label = 'phase'
+        period_plot.yaxis.axis_label = 'mag'
+        period_plot.toolbar.logo = None
+
+        # do we have a distance modulus (dm)?
+        obj = DBSession().query(Obj).get(obj_id)
+        if obj.dm is not None:
+            period_plot.extra_y_ranges = {
+                "Absolute Mag": Range1d(start=ymax - obj.dm, end=ymin - obj.dm)
+            }
+            period_plot.add_layout(
+                LinearAxis(y_range_name="Absolute Mag", axis_label="m - DM"), 'right'
+            )
+
+        # initiate hover tool
+        period_imhover = HoverTool(tooltips=tooltip_format)
+        period_imhover.renderers = []
+        period_plot.add_tools(period_imhover)
+
+        # store all the plot data
+        period_model_dict = {}
+
+        # iterate over each filter
+        for i, (label, df) in enumerate(split):
+
+            # fold x-axis on period in days
+            df['mjd_folda'] = (df['mjd'] % period) / period
+            df['mjd_foldb'] = df['mjd_folda'] + 1.0
+
+            # phase a plotting
```

Comment:
Seems like you could make this section more compact by iterating over `a` and `b` to plot the points/errorbars.

Reply:
I admit I am not very familiar with javascript and I don't quite know how to implement this suggestion over both python and javascript.  I will experiment with this.

Reply:
Got it.  The most recent commit should address this.